### ISSUE TITLE
regression: _zsh_highlight_load_highlighters:12: bad pattern

### ISSUE THREAD

**Issue by nicoulaj** [zsh-users/zsh-syntax-highlighting#535]
[This commit](https://github.com/zsh-users/zsh-syntax-highlighting/commit/db6cac391bee957c20ff3175b2f03c4817253e60) introduced a regression for me, z-sy-h won't load anymore:
```
_zsh_highlight_load_highlighters:12: bad pattern: /home/nicoulaj/work/zsh-syntax-highlighting/highlighters/*/(/)
```

Probably some `localoptions` are required ?

My options:
```
$ zsh --version
zsh 5.6.2 (x86_64-pc-linux-gnu)
$ setopt
autocd
autocontinue
noautomenu
autonamedirs
autopushd
autoresume
nobareglobqual
nobeep
nocaseglob
cbases
cdablevars
chaselinks
combiningchars
completeinword
correct
emacs
extendedglob
extendedhistory
globcomplete
nohashcmds
nohashdirs
nohistbeep
histexpiredupsfirst
histfindnodups
histignoredups
histreduceblanks
histsavenodups
histsubstpattern
histverify
incappendhistory
interactive
interactivecomments
nolistbeep
listpacked
longlistjobs
monitor
pathdirs
printeightbit
promptsubst
pushdignoredups
pushdtohome
rcexpandparam
```

bisect log:
```
$ git bisect start                                                                                                             master 
$ git bisect bad HEAD                                                                                                 master (bisect) 
$ git bisect good 0.6.0                                                                                               master (bisect) 
Bisecting: 51 revisions left to test after this (roughly 6 steps)
[8652a8f4e4c1a2d8fd36b4ee46515b42ad09c97f] Bump copyright years.
$ git fetch origin                                                                                   remotes/origin/pr/473~3 (bisect) 
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                            remotes/origin/pr/473~3 (bisect) 
Bisecting: 25 revisions left to test after this (roughly 5 steps)
[7b417be1ce29c549052e989ef2b628a0de0d0a79] tests: Escape newlines in TAP description
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                            remotes/origin/pr/492~7 (bisect) 
Bisecting: 12 revisions left to test after this (roughly 4 steps)
[2c0c75a8ee9a0f212eebbe4ee91ca5884f18e52f] driver, main: Add -z to autoload calls
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                            remotes/origin/pr/508~2 (bisect) 
Bisecting: 6 revisions left to test after this (roughly 3 steps)
[5b539663c0d740a0c00169d5ecbd58e47ff16252] Add a regression test for a problem seen during development.
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                            remotes/origin/pr/517~1 (bisect) 
Bisecting: 3 revisions left to test after this (roughly 2 steps)
[db6cac391bee957c20ff3175b2f03c4817253e60] main: Add glob qualifier to work around zsh/macOS bug
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                              remotes/origin/pr/518 (bisect) 
_zsh_highlight_load_highlighters:12: bad pattern: /home/nicoulaj/work/zsh-syntax-highlighting/highlighters/*/(/)
Bisecting: 0 revisions left to test after this (roughly 1 step)
[ee26d44de1d1ace0b7a8680e21396f8175d61f6b] Merge pull request #517 from tbroadley/fix-typos
$ source zsh-syntax-highlighting.zsh && git bisect good || git bisect bad                                       bisect/bad~1 (bisect) 
db6cac391bee957c20ff3175b2f03c4817253e60 is the first bad commit
commit db6cac391bee957c20ff3175b2f03c4817253e60
Author: Bengt Brodersen <bengt.brodersen@gmail.com>
Date:   Wed May 30 19:03:36 2018 +0200

    main: Add glob qualifier to work around zsh/macOS bug
    
    zsh/macOS bug: http://www.zsh.org/cgi-bin/mla/redirect?WORKERNUMBER=42891

:100644 100644 9ffde7622337d31e286af62aa690ad2ef36ae1c9 6838e1d741475a642f3592fad57f887898f82946 M      zsh-syntax-highlighting.zsh
```

**Comment by danielshahaf** [comment]
NO_BARE_GLOB_QUAL is the setopt that breaks this.

The broken commit was #518.

/cc @qoomon FYI

**Comment by nicoulaj** [comment]
Pushed a commit that enables `BARE_GLOB_QUAL ` locally.
@qoomon can you please test it doesn't break the fix for #518 ?

**Comment by qoomon** [comment]
Still working
